src/util/api.test.js

```diff
@@ -0,0 +1,279 @@
+                'https://rickandmortyapi.com/api/character/343',
+                'https://rickandmortyapi.com/api/character/356',
+                'https://rickandmortyapi.com/api/character/394',
+              ],
+              url: 'https://rickandmortyapi.com/api/location/1',
+              created: '2017-11-10T12:42:04.162Z',
+            },
+          ],
+        }),
+      };
+    }
+    case 'https://rickandmortyapi.com/api/episode?page=1': {
+      return {
+        ok: false,
+        status: 404,
+        json: async () => ({ error: 'There is nothing here' }),
+      };
+    }
+    case 'https://rickandmortyapi.com/api/character/20': {
+      return {
+        ok: true,
+        status: 200,
+        json: async () => ({
+          id: 20,
+          name: 'Ants in my Eyes Johnson',
+          status: 'unknown',
+          species: 'Human',
+          type: 'Human with ants in his eyes',
+          gender: 'Male',
+          origin: { name: 'unknown', url: '' },
+          location: { name: 'Interdimensional Cable', url: 'https://rickandmortyapi.com/api/location/6' },
+          image: 'https://rickandmortyapi.com/api/character/avatar/20.jpeg',
+          episode: ['https://rickandmortyapi.com/api/episode/8'],
+          url: 'https://rickandmortyapi.com/api/character/20',
+          created: '2017-11-04T22:34:53.659Z',
+        }),
+      };
+    }
+    case 'https://rickandmortyapi.com/api/location/5': {
+      return {
+        ok: true,
+        status: 200,
+        json: async () => ({
+          id: 5,
+          name: 'Anatomy Park',
+          type: 'Microverse',
+          dimension: 'Dimension C-137',
+          residents: [
+            'https://rickandmortyapi.com/api/character/12',
+            'https://rickandmortyapi.com/api/character/17',
+            'https://rickandmortyapi.com/api/character/96',
+            'https://rickandmortyapi.com/api/character/97',
+            'https://rickandmortyapi.com/api/character/98',
+            'https://rickandmortyapi.com/api/character/99',
+            'https://rickandmortyapi.com/api/character/100',
+            'https://rickandmortyapi.com/api/character/101',
+            'https://rickandmortyapi.com/api/character/108',
+            'https://rickandmortyapi.com/api/character/268',
+            'https://rickandmortyapi.com/api/character/300',
+          ],
+          url: 'https://rickandmortyapi.com/api/location/5',
+          created: '2017-11-10T13:08:46.060Z',
+        }),
+      };
+    }
+    case 'https://rickandmortyapi.com/api/episode/200': {
+      return {
+        ok: false,
+        status: 404,
+        json: async () => ({ error: 'Episode not found' }),
+      };
+    }
+    default: {
+      throw new Error(`Unmocked API endpoint: ${url}`);
+    }
+  }
+};
+
+describe('API fetching tests', () => {
+  beforeEach(() => {
+    window.fetch = jest.fn().mockImplementation(fetchMock);
+  });
+
+  afterEach(() => {
+    // @ts-ignore
+    global.fetch.mockClear();
```

Comment:
Raz używasz `window` raz `global`, a tak naprawdę to ten sam object, więc wybierz `window` jeśli testujesz z wykorzystaniem [jsdom](https://jestjs.io/docs/configuration#testenvironment-string). Swoją drogą można pominąć `afterEach` i dodać do jest.config.js [resetMocks: true](https://jestjs.io/docs/configuration#resetmocks-boolean).